src/modules/ie-objects/helpers/convert-node-to-es-query-filter-objects.ts

```diff
@@ -0,0 +1,90 @@
+import { BadRequestException } from '@nestjs/common';
+import { baseTypes, Expression } from 'jsep';
+
+import { SearchFilter } from '../dto/ie-objects.dto';
+
+export const convertNodeToEsQueryFilterObjects = (
+	node: Expression,
+	searchTemplate?: any[],
+	searchFilter?: SearchFilter
+): any => {
+	switch (node.type) {
+		case 'Compound':
+			return {
+				bool: {
+					should: ((node.body || []) as Expression[]).map((bodyNode) =>
+						convertNodeToEsQueryFilterObjects(bodyNode, searchTemplate, searchFilter)
+					),
+				},
+			};
+
+		case 'SequenceExpression':
+			return {
+				bool: {
+					should: ((node.expressions || []) as Expression[]).map((bodyNode) =>
+						convertNodeToEsQueryFilterObjects(bodyNode)
+					),
+				},
+			};
```

Comment:
Both should also have a minimum_should_match equal to the number of elements in the array, otherwise this isn't an AND statements but instead an OR statement